client/lib/warn/index.js

```diff
@@ -1,14 +1,13 @@
 /**
  * Internal Dependencies
  */
-var config = require( 'config' );
+import config from 'config';
 
-function warn() {
-	if ( config( 'env' ) !== 'production' ) {
```

Comment:
We can use also `process.env.NODE_ENV !== 'production'` and the block should be removed during compilation.